js/modules/k6/experimental/streams/readable_streams_test.go

```diff
@@ -0,0 +1,123 @@
+//go:build wpt
+
+package streams
+
+import (
+	"testing"
+
+	"go.k6.io/k6/js/modules"
+	"go.k6.io/k6/js/modules/k6/timers"
+	"go.k6.io/k6/js/modulestest"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestReadableStream(t *testing.T) {
+	t.Parallel()
+
+	suites := []string{
+		"bad-strategies.any.js",
+		"bad-underlying-sources.any.js",
+		"cancel.any.js",
+		"constructor.any.js",
+		"count-queuing-strategy-integration.any.js",
+		"default-reader.any.js",
+		"floating-point-total-queue-size.any.js",
+		"general.any.js",
+		"reentrant-strategies.any.js",
+		"templated.any.js",
+	}
+
+	for _, s := range suites {
+		s := s
+		t.Run(s, func(t *testing.T) {
+			t.Parallel()
+			ts := newConfiguredRuntime(t)
+			gotErr := ts.EventLoop.Start(func() error {
+				return executeTestScripts(ts.VU, "tests/wpt/streams/readable-streams", s)
+			})
+			assert.NoError(t, gotErr)
+		})
+	}
+}
+
+func newConfiguredRuntime(t testing.TB) *modulestest.Runtime {
+	rt := modulestest.NewRuntime(t)
+
+	// We want to make the [self] available for Web Platform Tests, as it is used in test harness.
+	_, err := rt.VU.Runtime().RunString("var self = this;")
+	require.NoError(t, err)
+
+	// We also want to make [timers.Timers] available for Web Platform Tests.
+	for k, v := range timers.New().NewModuleInstance(rt.VU).Exports().Named {
+		require.NoError(t, rt.VU.RuntimeField.Set(k, v))
+	}
+
+	// We also want the streams module exports to be globally available.
+	m := new(RootModule).NewModuleInstance(rt.VU)
+	for k, v := range m.Exports().Named {
+		require.NoError(t, rt.VU.RuntimeField.Set(k, v))
+	}
+
+	// Then, we register the Web Platform Tests harness.
+	compileAndRun(t, rt, "tests/wpt", "resources/testharness.js")
+
+	// And the Streams-specific test utilities.
+	files := []string{
+		"resources/rs-test-templates.js",
+		"resources/rs-utils.js",
+		"resources/test-utils.js",
+	}
+	for _, file := range files {
+		compileAndRun(t, rt, "tests/wpt/streams", file)
+	}
+
+	return rt
+}
+
+func compileAndRun(t testing.TB, runtime *modulestest.Runtime, base, file string) {
+	program, err := modulestest.CompileFile(base, file)
+	require.NoError(t, err)
+
+	_, err = runtime.VU.Runtime().RunProgram(program)
+	require.NoError(t, err)
+}
+
+func executeTestScripts(vu modules.VU, base string, scripts ...string) error {
+	for _, script := range scripts {
```

Comment:
I was very confused by this taking of multiple scripts , and why we are runnign them in the same runtime .. until I figured out we never run multiple scrtipts.

Can this be removed ?

just so somebody doesn't try to use it and then to try to figureo out why it isn't working.